internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -696,10 +576,230 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
+		reason = ClusterReadyReason
+		msg = ClusterReadyMsg
+	case models.ClusterStatusInsufficient, models.ClusterStatusPendingForInput:
+		condStatus = corev1.ConditionFalse
+		reason = ClusterNotReadyReason
+		msg = ClusterNotReadyMsg
+	case models.ClusterStatusPreparingForInstallation, models.ClusterStatusInstalled,
+		models.ClusterStatusInstalling, models.ClusterStatusInstallingPendingUserAction,
+		models.ClusterStatusError, models.ClusterStatusAddingHosts, models.ClusterStatusFinalizing:
+		condStatus = corev1.ConditionFalse
+		reason = ClusterAlreadyInstallingReason
+		msg = ClusterAlreadyInstallingMsg
+	default:
+		condStatus = corev1.ConditionUnknown
+		reason = v1beta1.UnknownStatusReason
+		msg = fmt.Sprintf("%s %s", v1beta1.UnknownStatusMsg, status)
+	}
+	setStatusCondition(&cluster.Status.Conditions, hivev1.ClusterDeploymentCondition{
+		Type:    ClusterReadyForInstallationCondition,
+		Status:  condStatus,
+		Reason:  reason,
+		Message: msg,
+	})
+}
+
+func clusterInstalled(cluster *hivev1.ClusterDeployment, status, statusInfo string) {
+	var condStatus corev1.ConditionStatus
+	var reason string
+	var msg string
+	switch status {
+	case models.ClusterStatusInstalled:
+		condStatus = corev1.ConditionTrue
+		reason = v1beta1.InstalledReason
+		msg = fmt.Sprintf("%s %s", v1beta1.InstalledMsg, statusInfo)
+	case models.ClusterStatusError:
+		condStatus = corev1.ConditionFalse
+		reason = v1beta1.InstallationFailedReason
+		msg = fmt.Sprintf("%s %s", v1beta1.InstallationFailedMsg, statusInfo)
+	case models.ClusterStatusInsufficient, models.ClusterStatusPendingForInput, models.ClusterStatusReady:
+		condStatus = corev1.ConditionFalse
+		reason = v1beta1.InstallationNotStartedReason
+		msg = v1beta1.InstallationNotStartedMsg
+	case models.ClusterStatusPreparingForInstallation, models.ClusterStatusInstalling, models.ClusterStatusFinalizing,
+		models.ClusterStatusInstallingPendingUserAction:
+		condStatus = corev1.ConditionFalse
+		reason = v1beta1.InstallationInProgressReason
+		msg = fmt.Sprintf("%s %s", v1beta1.InstallationInProgressMsg, statusInfo)
+	default:
+		condStatus = corev1.ConditionUnknown
+		reason = v1beta1.UnknownStatusReason
+		msg = fmt.Sprintf("%s %s", v1beta1.UnknownStatusMsg, status)
+	}
+	setStatusCondition(&cluster.Status.Conditions, hivev1.ClusterDeploymentCondition{
+		Type:    ClusterInstalledCondition,
+		Status:  condStatus,
+		Reason:  reason,
+		Message: msg,
+	})
+}
+
+func clusterValidated(clusterDeployment *hivev1.ClusterDeployment, status string, c *common.Cluster) {
+	failedValidationInfo := ""
+	validationRes, err := cluster.GetValidations(c)
+	var failures []string
+	if err == nil {
+		for _, vRes := range validationRes {
+			for _, v := range vRes {
+				if v.Status == cluster.ValidationFailure {
+					failures = append(failures, v.Message)
+				}
+			}
+		}
+		failedValidationInfo = strings.Join(failures[:], ",")
+	}
+	var condStatus corev1.ConditionStatus
+	var reason string
+	var msg string
+	switch {
```

Comment:
Maybe `ClusterStatusPendingForInput` belongs here as well? 
in this case some of the validations will fail and this is the way for the user to know it.

Reply:
@avishayt WDYT?

Reply:
How about Unknown with a different message?

Reply:
Sure. What would be the message?